### PR TITLE
crimson/monc: renew tickets and rotating keys on MAuthReply

### PR DESCRIPTION
**Pull request by rzarzynski**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/mon/MonClient.cc [hunk 1]
```diff
@@ -156,9 +156,14 @@ seastar::future<> Connection::renew_rotating_keyring()
...
   auto ttl = std::chrono::seconds{
     static_cast<long>(crimson::common::local_conf()->auth_service_ticket_ttl)};
-  auto cutoff = now - ttl / 4;
-  if (!rotating_keyring->need_new_secrets(utime_t(cutoff))) {
+  auto cutoff = utime_t{now - std::min(std::chrono::seconds{30}, ttl / 4)};
+  if (!rotating_keyring->need_new_secrets(cutoff)) {
+    logger().debug("renew_rotating_keyring secrets are up-to-date "
+                   "(they expire after {})", cutoff);
     return seastar::now();
+  } else {
```
Comment (ronen-fr) [inline src/crimson/mon/MonClient.cc:164]: not sure we need the 'else' block as a block here.

Comment (rzarzynski) [reply src/crimson/mon/MonClient.cc:164]: It's here to align visually with the debug above.

**Comment by ronen-fr** [comment]
Is there a point in having 'crimson/monc: renew tickets and rotating keys on MAuthReply' as a separate commit? for me - it made reading slightly harder

**Comment by rzarzynski** [comment]
> Is there a point in having 'crimson/monc: renew tickets and rotating keys on MAuthReply' as a separate commit? for me - it made reading slightly harder

Just the separation between fixes and cleanups.

File: src/crimson/mon/MonClient.cc [hunk 2]
```diff
@@ -816,7 +826,11 @@ seastar::future<> Client::handle_auth_reply(crimson::net::ConnectionRef conn,
   if (found != pending_conns.end()) {
     return (*found)->handle_auth_reply(m);
   } else if (active_con) {
-    return active_con->handle_auth_reply(m);
+    return active_con->handle_auth_reply(m).then([this] {
+      return seastar::when_all_succeed(
+        active_con->renew_rotating_keyring(),
+        active_con->renew_tickets()).then_unpack([] {});
```
Comment (tchaikov) [inline src/crimson/mon/MonClient.cc:832]:
```suggestion
        active_con->renew_tickets())..discard_result();
```

**Review by tchaikov — APPROVED** [review]

File: src/crimson/mon/MonClient.cc (hunk 2 above)
Comment (rzarzynski) [reply src/crimson/mon/MonClient.cc:832]: I guess this can be applied in other places too. If so, will address as a follow up / separate commit.

Comment (tchaikov) [reply src/crimson/mon/MonClient.cc:832]: yup. we can have a follow-up PR to implement this cleanup.

Comment (rzarzynski) [reply src/crimson/mon/MonClient.cc:832]: Sent as https://github.com/ceph/ceph/pull/47285.

**Pull request merged**